### PR TITLE
 homework week3 Ali Amouri Kadhim

### PR DESCRIPTION
**Pull request by AliKdhim87**

### CODE REVIEW

File: Week3/homework/addToShoppingCart.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+'use strict';
+function addToShoppingCart(item) {
+    // An array with two Items
+    var array = ['bananas', 'milk'];
```
Comment (nudge) [inline Week3/homework/addToShoppingCart.js]: 1. Use `let`
2. Use a more descriptive name than `array`, perhaps `shoppingCart` works better.

File: Week3/homework/addToShoppingCart.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    if (array.length >= 1) {
+        // With shift method i can remove the first item from the array
+        array.shift()
+    }
+    return `You bought ${array} !`;
+
+}
+console.log(addToShoppingCart('Tomato')); //Output You bought milk,Tomato !
+console.log(addToShoppingCart('Water')); //You bought milk,Water ! 
+console.log(addToShoppingCart('Orange')); //You bought milk,Orange !
```
Comment (nudge) [inline Week3/homework/addToShoppingCart.js]: The question is asking that you can hold maximum 3 items in the shopping cart, so the expected output should be:
```
You bought bananas,milk,Tomato !
You bought milk,Tomato,Water !
You bought Tomato,Water,Orange !
```
See if you can update this code to fix this. Hint: Update the check of the array length!

File: Week3/homework/calculateDogAge.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+'use strict';
+function calculateDogAge(puppysAge) {
+    const dogYears = 7 * puppysAge;
+    return console.log(`Your doggie is${dogYears} years old in dog years!`);
```
Comment (nudge) [inline Week3/homework/calculateDogAge.js]: The question asks to return a string here.

See if you can update the code to fix this. Hint: Move the `console.log` outside this function.

File: Week3/homework/calculateTotalPrice.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+'use strict';
+const calculateTotalPrice = (object) => {
+    let sum = null;
```
Comment (nudge) [inline Week3/homework/calculateTotalPrice.js]: nit: might be clearer to do `let sum = 0;`

File: Week3/homework/calculateTotalPrice.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    return sum;
+};
+const cartForParty = {
+    juice: 1.25,
+    chips: 2.25,
+    cola: 1,
+    water: 0.9,
+    chhocolateBar: 1.75
+};
+console.log(calculateTotalPrice(cartForParty));
```
Comment (nudge) [inline Week3/homework/calculateTotalPrice.js]: 👍

File: Week3/homework/giveCompliment.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+'use strict';
+const giveCompliment = (name) => {
+    // Array with ten Complimen
+    const complimentArray = ['great', 'awesome', 'great', 'nice', 'Elegant', 'Elegant', 'Majestic', 'Lovely', 'Striking', 'Amazing'];
+    // to ronden the content of the array 
+    const randomArray = Math.floor(Math.random() * complimentArray.length);
```
Comment (nudge) [inline Week3/homework/giveCompliment.js]: nit: `randomIndex` might be a clearer name

File: Week3/homework/giveCompliment.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+
+    return `You are  ${complimentArray[randomArray]} ${name}`
+}
+console.log(giveCompliment('Ali'));
+console.log(giveCompliment('Ali'));
+console.log(giveCompliment('Ali'));
```
Comment (nudge) [inline Week3/homework/giveCompliment.js]: 👍

File: Week3/homework/tellFortune.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+    const rondomLoc = Math.floor(Math.random() * location.length);
+    const rondomJob = Math.floor(Math.random() * jobTitle.length);
+    return console.log(`You will be a ${jobTitle[rondomJob]} in ${location[rondomLoc]}, and married to ${partners[rondomPar]} with ${children[rondomChi]} kids`)
+}
+const numChildren = [1, 2, 3, 4];
+const partnerNames = ['Eva', 'Emma', 'Anna', 'Tess', 'Mila'];
+const locations = ['Irak', 'Netherland', 'Brazil', 'Gernany', 'Turkey'];
+const jobs = ['Front-End developer', 'Back-end developer', 'Full-Stack developer', 'Sport coach', 'Policeman'];
+
+tellFortune(numChildren, partnerNames, locations, jobs);
```
Comment (nudge) [inline Week3/homework/tellFortune.js:15]: 👍  nice! Great use of string templates.

File: Week3/homework/creditCardValidator.js [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+        let currentNumber = creditCardNum[i];
+
+        // turn the digit from a string to an integer (if the digit is in fact a digit and not anther char)
+        currentNumber = Number.parseInt(currentNumber);
+
+        // check that the digit is a number
+        if (!Number.isInteger(currentNumber)) {
+            return false;
+        }
+    }
```
Comment (nudge) [inline Week3/homework/creditCardValidator.js]: Your logic works which is great.
Another way to do this more compactly is using `Number()` object, see: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number
It tries to parse the entire string and see if it is a number, and if not returns `NaN`.

You could do a check like `Number.isInteger(Number(creditCardNum))`

File: Week3/homework/creditCardValidator.js [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+    return true;
+};
+
+/**** tests *****/
+console.log(validateCreditCard('9999777788880000')); //true
+console.log(validateCreditCard('6666666666661666')); //true
+console.log(validateCreditCard('a92332119c011112')); //false
+console.log(validateCreditCard('4444444444444444')); //false
+console.log(validateCreditCard('1211111111111112')); //true
```
Comment (nudge) [inline Week3/homework/creditCardValidator.js]: 👍 nicely done!!

File: Week3/homework/tellFortune.js (hunk 8 above)
Comment (AliKdhim87) [reply Week3/homework/tellFortune.js:15]: Thank you very much for the feedback

**Review by AliKdhim87 — COMMENTED** [review]
I resolved all this problem.

File: Week3/homework/addToShoppingCart.js (hunk 1 above)
Comment (AliKdhim87) [reply Week3/homework/addToShoppingCart.js]: Thank you for the feed back I fixed

File: Week3/homework/addToShoppingCart.js (hunk 2 above)
Comment (AliKdhim87) [reply Week3/homework/addToShoppingCart.js]: Thank you for the feed back I fixed

File: Week3/homework/calculateDogAge.js (hunk 3 above)
Comment (AliKdhim87) [reply Week3/homework/calculateDogAge.js]: Thank you for the feed back I fixed.

File: Week3/homework/calculateTotalPrice.js (hunk 4 above)
Comment (AliKdhim87) [reply Week3/homework/calculateTotalPrice.js]: Thank you for the feedback I fixed

File: Week3/homework/calculateTotalPrice.js (hunk 5 above)
Comment (AliKdhim87) [reply Week3/homework/calculateTotalPrice.js]: Thank you for the feedback

File: Week3/homework/giveCompliment.js (hunk 6 above)
Comment (AliKdhim87) [reply Week3/homework/giveCompliment.js]: Thank you for the feed back I changed the name

File: Week3/homework/giveCompliment.js (hunk 7 above)
Comment (AliKdhim87) [reply Week3/homework/giveCompliment.js]: Thank you for the feedback

**Pull request closed without merging**